### PR TITLE
Warn or error if incompatible park version is loaded

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Can you also add a warning for when the park file was made at too low a version and add a minimum supported version warning for that. Also we really need to rename minversion as it's not the minimum version supported but the min forward compatible version luckily forward compatible version will 99% of the time match target version.

**Comment by IntelOrca** [comment]
> Can you also add a warning for when the park file was made at too low a version and add a minimum supported version warning for that.

How do we know what is too low?

**Comment by duncanspumpkin** [comment]
> > Can you also add a warning for when the park file was made at too low a version and add a minimum supported version warning for that.
> 
> How do we know what is too low?

Well we have a constant for the minimum version same as the constant for the minimum forward compatible version.

File: src/openrct2/park/ParkFile.h [hunk 1]
```diff
@@ -13,6 +13,9 @@ namespace OpenRCT2
     // The minimum version that is forwards compatible with the current version.
     constexpr uint32_t PARK_FILE_MIN_VERSION = 0x9;
 
+    // The minimum version that is backwards compatible with the current version.
+    constexpr uint32_t PARK_FILE_MIN_SUPPORTED_VERSION = 0x1;
```
Comment (Gymnasiast) [inline src/openrct2/park/ParkFile.h:17]: Did we start at 0x1, or at 0x0?

Comment (IntelOrca) [reply src/openrct2/park/ParkFile.h:17]: 0, but I can't use 0 because the compilers complain that I am doing 0 < 0.

Comment (Gymnasiast) [reply src/openrct2/park/ParkFile.h:17]: Sure, but this means that .park files that loaded correctly before will not load now. There has to be a way to satisfy the compilers without cutting off v0 support so quickly.

Comment (Broxzier) [reply src/openrct2/park/ParkFile.h:17]: Sounds like this check is useless for now, until we drop support for older versions.

Comment (Gymnasiast) [reply src/openrct2/park/ParkFile.h:17]: I think we could do two things:

- Leave out the check for "too old" parks, or
- Wrap the code in an `if (PARK_FILE_MIN_SUPPORTED_VERSION > 0)` to satisfy the compilers.

Comment (IntelOrca) [reply src/openrct2/park/ParkFile.h]: I have only done this to satisfy @duncanspumpkin's request. Check with him.
I haven't actually tested whether v0 parks still load correctly, maybe they don't?

Comment (duncanspumpkin) [reply src/openrct2/park/ParkFile.h]: I wouldn't expect v0 parks to load. There was a whole bunch of issues in like v3.

**Comment by Gymnasiast** [comment]
I have decided to merge as-is. It will now at least show a nice message if the save is too new, which I expect will be more common in the longer term anyway.

The minimum version requires more testing anyway, and even if v0/v1/v2 parks do not load, my preference would be to see if we can fix those. Failing that, we can change the minimum to v3 later.

**Pull request merged**